nodes/persistence-node/src/modules/daemon/daemon.model.ts

```diff
@@ -0,0 +1,43 @@
+import { HttpConfig } from "../../api-server/HttpConfig";
+import { HttpsConfig } from "../../api-server/HttpsConfig";
+import { buildMainDependencyContainer, MainDependencyContainer } from "./daemon.deps";
+
+export class Daemon {
+
+    private constructor(
+        private deps: MainDependencyContainer,
+        shouldLog: boolean
+    ) {
+        this.deps.persistenceNodeApi.run()
+        this.deps.loggerConfig.shouldLog = shouldLog;
+     }
+
+    static async build(shouldLog: boolean): Promise<Daemon> {
+        const container = await buildMainDependencyContainer();
+
+        return new Daemon(container.cradle, shouldLog);
+    }
+
+    async runApi(httpConfig: HttpConfig, httpsConfig: HttpsConfig) {
```

Comment:
We can maybe rename this to "run" and have it start both the persistence API, the ipfsGatewayApi and listenForEvents

Reply:
Great idea, I've implemented it.